third_party/move/move-compiler-v2/src/inliner.rs

```diff
@@ -1140,6 +1140,14 @@ impl<'env, 'rewriter> ExpRewriterFunctions for InlinedRewriter<'env, 'rewriter>
         }
     }
 
+    fn internal_rewrite_pattern(&mut self, pat: &Pattern, entering_scope: bool) -> (bool, Pattern) {
```

Comment:
I don't understand this. The internal_ functions are supposed to be not overriden, thats why the are called internal. Also this kind of deactivates recursive rewriting.

Is it perhaps that we need to make rewrite_pattern non-recursive as we do the traversal now in the rewriter?